### ISSUE TITLE
Async Javascript

### ISSUE THREAD

**Issue by mhfrantz** [hiptest/hiptest-publisher#17]
The JS code is generated for a synchronous interface.  However, most JS is asynchronous, and adapting is awkward.

The two major async styles are callbacks and promises.  ES6 defines the promise standard.

**Comment by vincent-psarga** [comment]
The thing is that we want to generate generic enough code, so it fits most needs.
Depending on the library used for testing, the code generated by hiptest-publisher works fine with asynchronous calls (for example, using Ember.js test helpers can be used with hiptest-publisher generated code).

That said, we'll keep that in mind.

**Comment by mhfrantz** [comment]
If you want "generic enough code, so it fits most needs," then you should generate some sort of asynchronous interface, since calling a synchronous API from an asynchronous function is easily handled, while calling an asynchronous API from a synchronous function is not.  Perhaps by "most needs" you mean "certain web development frameworks that provide a synchronous facade" (Ember, Protractor)?

I'd be happy to contribute alternate code generation modules, e.g. "mocha-promises", if you would accept expansion in that direction.  That would make your tool applicable to a wider range of testing situations.

**Comment by vincent-psarga** [comment]
Indeed. We'll discuss that with the team but we're always glad when people want to contribute :)
I'd be really interested on how would the action words/scenarios look when generated for mocha-promises. I think it'd be nice to have a "new" export type specifically for mocha with promises :)

**Comment by mhfrantz** [comment]
With promises, the assumption would be that each of the action word function would return a promise.  Thus, the stubs are identical to what you are generating now for the synchronous case.  However, the difference comes in the scenarios, e.g. Mocha:
```
it('should do something', function () {
  return this.actionwords.oneThing()
    .then(() => this.actionwords.anotherThing())
    .then(() => this.actionwords.somethingWithParameters('foo'));
});
```
